Add Check Invalid Rotation Plugin

PlanesZwalker opened the pull request:
- Detects non-standard rotations in models before export
- Shows warning dialog with element names and suggested fixes
- Auto-fix functionality for rotations to nearest standard values
- User choice to continue export or cancel to fix first
- Supports multiple rotation fix options
- Compatible with Blockbench 4.12.6+ and Minecraft Bedrock standards

PlanesZwalker committed:
Add Rotation Checker Plugin

- Automatically detects non-standard rotations in models
- Shows warning dialog before export to prevent Minecraft display issues
- Auto-fix functionality to correct rotations to nearest standard values
- User choice to continue export or cancel to fix rotations first
- Compatible with Blockbench 4.12.6+

PlanesZwalker committed:
Add demo.png to rotation checker plugin

JannisX11 commented on plugins/check_invalid_rotation/check_invalid_rotation.js:
```diff
@@ -0,0 +1,700 @@
+        icon: 'warning',
+        version: '1.0.0',
+        variant: 'both',
+        tags: ['minecraft', 'rotation', 'validation', 'export'],
```
You can specify up to 3 tags. Make sure they are formatted for display in UI, in this case they should start with an upper case letter.

JannisX11 commented on plugins/check_invalid_rotation/check_invalid_rotation.js:
```diff
@@ -0,0 +1,700 @@
+        return nonStandardRotations;
+    }
+    
+    function applyRotationFix(rotation) {
```
This seems to be unused.

JannisX11 commented on plugins/check_invalid_rotation/check_invalid_rotation.js:
```diff
@@ -0,0 +1,700 @@
+                Outliner.update();
+            }
+            
+            if (Project && Project.update) {
```
Does not exist

JannisX11 commented on plugins/check_invalid_rotation/check_invalid_rotation.js:
```diff
@@ -0,0 +1,700 @@
+                Canvas.updateAll();
+            }
+            
+            if (Canvas && Canvas.render) {
```
Does not exist

JannisX11 commented on plugins/check_invalid_rotation/check_invalid_rotation.js:
```diff
@@ -0,0 +1,700 @@
+                Canvas.render();
+            }
+            
+            if (Timeline && Timeline.update) {
```
Does not exist

JannisX11 commented on plugins/check_invalid_rotation/changelog.json:
```diff
@@ -0,0 +1,18 @@
+[
```
The changelog file is malformatted.

JannisX11 commented on plugins/check_invalid_rotation/about.md:
```diff
@@ -0,0 +1,46 @@
+- Automatic rotation normalization
+- Smart duplicate detection
+
+*Version 1.0.0 - Compatible with Blockbench 4.12.6+*
```
Version and compatibility is already displayed on the plugin page, I would recomment not having it here again since it's always another step to change it when updating the plugin.

JannisX11 reviewed (changes requested):
Left comments on a few issues that I've found.

Apart from that, the issue that I noticed was that no check was performed when saving via Ctrl + S, or File > Save Model.

Btw, in case you didn't see, Minecraft recently lifted the 22.5 degree limit for models, and on bedrock it has never existed in the first place. So I assume this is just intended for legacy versions, any maybe for stylistic consistency if the style requirement is to align with these rotations?

Also, for anything validation-related I can also recomment the Validator API, it can display issues in real time in the UI so could be an addition: https://web.blockbench.net/docs/classes/custom_validator.ValidatorCheck.html

PlanesZwalker committed:
Fix Blockbench compatibility and improve documentation

- Remove non-compatible API calls from refreshScene() function
- Simplify about.md by removing duplicate version info and repetitive benefits
- Streamline changelog.json to avoid feature duplication
- Address feedback from @JannisX11 for better maintainability

PlanesZwalker committed:
Update rotation checker plugin for better accuracy

- Target specific version range (1.8 - 1.18.x) instead of broad 1.8+
- Add disclaimer about Minecraft 1.19+ potentially relaxing restrictions
- Update changelog format to match schema (object instead of array)
- Reduce tags to 3 maximum (removed 'minecraft' tag)
- Improve version specificity and honesty about plugin scope
- Update both English and French translations consistently

PlanesZwalker committed:
Fix rotation checker plugin: remove duplicate dialogs, clean up code, fix uninstall cleanup

- Removed duplicate save integration methods causing multiple dialogs
- Cleaned up code for production (removed comments, optimized formatting)
- Fixed uninstall cleanup to properly restore BarItems functions
- Individual 'Corriger' buttons now work correctly
- Reduced file size by 24% (1257 → 860 lines)
- All functionality tested and working

PlanesZwalker committed:
Merge branch 'JannisX11:master' into rotation-checker-plugin

PlanesZwalker committed:
Update check_invalid_rotation: improve about.md and add plugins.json entry

PlanesZwalker committed:
Merge branch 'rotation-checker-plugin' of https://github.com/PlanesZwalker/blockbench-plugins-Translation-planes-gizmo into rotation-checker-plugin
# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

PlanesZwalker committed:
Update check_invalid_rotation plugin: clean code and improve documentation

JannisX11 committed:
Merge branch 'master' into pr/PlanesZwalker/785
